File: aide/pkg/grammar/testdata/sample.rb
# Sample Ruby script for grammar testing.

module Greeter
  def self.greet(name)
    puts "Hello, #{name}!"
  end
end

class Calculator
  def initialize(precision = 2)
    @precision = precision
  end

  def factorial(n)
    return 1 if n <= 1
    n * factorial(n - 1)
  end

  def fibonacci(n)
    return n if n <= 1
    fibonacci(n - 1) + fibonacci(n - 2)
  end

  def format(value)
    value.round(@precision)
  end
end

def main
  Greeter.greet("world")
  calc = Calculator.new
  puts "5! = #{calc.factorial(5)}"
  puts "fib(10) = #{calc.fibonacci(10)}"
end

main

File: aide/pkg/grammar/testdata/sample.lua
-- Sample Lua module for grammar testing.

local M = {}

local function fibonacci(n)
    if n <= 1 then
        return n
    end
    return fibonacci(n - 1) + fibonacci(n - 2)
end

local function factorial(n)
    if n == 0 then
        return 1
    end
    return n * factorial(n - 1)
end

function M.greet(name)
    print("Hello, " .. name)
end

function M.compute(n)
    return {
        fib = fibonacci(n),
        fact = factorial(n),
    }
end

return M
